src/compiler/sexp.rs

```diff
@@ -674,68 +674,54 @@ fn parse_sexp_step(loc: Srcloc, p: &SExpParseState, this_char: u8) -> SExpParseR
     }
 }
 
-fn parse_sexp_inner(
+fn parse_sexp_inner<I>(
     start_: Srcloc,
     p_: SExpParseState,
-    n_: usize,
-    s: &[u8],
-) -> Result<Vec<Rc<SExp>>, (Srcloc, String)> {
+    s: I,
+) -> Result<Vec<Rc<SExp>>, (Srcloc, String)>
+where
+    I: Iterator<Item = u8>,
+{
     let mut start = start_;
-    let mut parse_state = p_;
-    let mut char_index = n_;
+    let mut p = p_;
```

Comment:
```suggestion
    let mut parse_state = p_;
```

Reply:
hmm ya.

Reply:
you can just make `p_` mutable instead, assuming you need it to be

Reply:
But `p_` is an argument

Reply:
Ya can do.  I dislike that from a stylistic perspective, but rust wins.

Reply:
I'd prefer to not have mutable arguments in general.